crates/re_log_types/src/data_cell.rs

```diff
@@ -469,15 +510,34 @@ impl std::fmt::Display for DataCell {
 // ---
 
 impl DataCell {
-    /// Returns the total (heap) allocated size of the array in bytes.
+    /// Compute and cache the total (heap) allocated size of the underlying arrow array in bytes.
+    /// This does nothing if the size has already been computed and cached before.
     ///
-    /// Beware: this is costly! Cache the returned value as much as possible.
-    pub fn size_bytes(&self) -> u64 {
-        let DataCellInner { name, values } = &*self.inner;
+    /// The caller must the sole owner of this cell, as this requires mutating an `Arc` under the
+    /// hood. Returns false otherwise.
+    ///
+    /// Beware: this is _very_ costly!
+    #[inline]
+    pub fn compute_size_bytes(&mut self) -> bool {
+        if let Some(inner) = Arc::get_mut(&mut self.inner) {
+            inner.compute_size_bytes();
+            return true;
+        }
+        false
+    }
+}
 
-        std::mem::size_of_val(name) as u64 +
-            // Warning: this is surprisingly costly!
-            arrow2::compute::aggregate::estimated_bytes_size(&**values) as u64
+impl DataCellInner {
+    /// Compute and cache the total (heap) allocated size of the underlying arrow array in bytes.
+    /// This does nothing if the size has already been computed and cached before.
+    ///
+    /// Beware: this is _very_ costly!
+    #[inline]
+    pub fn compute_size_bytes(&mut self) {
+        if self.values_size_bytes == 0 {
+            self.values_size_bytes =
+                arrow2::compute::aggregate::estimated_bytes_size(&*self.values) as u64;
```

Comment:
can the estimate ever be zero? would it make more sense to use `Option<u64>` or `Option<NonZerou64>` for `values_size_bytes`?

Reply:
yeah it definitely can now that it only measures the data and not the whole cell